reg/types.go

```diff
@@ -136,6 +148,8 @@ func (v virtual) as(s Spec) Register {
 		kind:  v.kind,
 		Width: Width(s.Size()),
 		mask:  s.Mask(),
+		// Non-breaking space for sorting.
+		allocAt: fmt.Sprintf("\u00A0- As%v() %s", s.String(), getFnNameFile(2)),
```

Comment:
This breaks allocation, hinting that an `==` is used on this type somewhere.

Reply:
I suspect the `possible   map[reg.Virtual][]reg.Physical` is the problem here or with `type Allocation map[Register]Physical`